test/e2e/domain_not_matching_base_test.go

```diff
@@ -18,7 +18,7 @@ import (
 
 func TestDomainNotMatchingBase(t *testing.T) {
 	t.Parallel()
-	if infraConfig.Status.Platform != configv1.AWSPlatformType {
+	if infraConfig.Status.PlatformStatus.Type != configv1.AWSPlatformType {
```

Comment:
The tests currently pass but `PlatformStatus` can be `nil`.

Reply:
[API reference](https://github.com/openshift/api/blob/51f399230d604fa013c2bb341040c4ad126e7309/config/v1/types_infrastructure.go#L274) states `This value will be synced with to the status.platform and status.platformStatus.type. So, AFAIU it won't be nil in practice as it's being synced with `status.platform`. To be on the safe side, I've added a nil check to [L#32](https://github.com/openshift/cluster-ingress-operator/pull/814/files#diff-0945d1d0d6ad7f06323a0a6169b3cdd00c6ed887a60aca879530bb1bb09a98beR32). But, for the tests, I think it is not needed.

Reply:
This is an e2e test that can be run on anything, so it's safer to just add a nil check.
```suggestion
	if infraConfig.Status.PlatformStatus != nil && infraConfig.Status.PlatformStatus.Type != configv1.AWSPlatformType {
```

Reply:
I added a nil check.